react/src/App.tsx

```diff
@@ -55,58 +56,85 @@ function BaseApp(props: { darkMode: boolean; toggleDarkMode: () => void }) {
         if (apiInfo?.oauth) {
             body = { redirect_uri: window.location.origin };
         }
-        const result = await apiFetch(`/api/logout`, {
-            method: "POST",
-            headers: { "Content-Type": "application/json" },
-            body: JSON.stringify(body),
-        });
-        if (result.ok) {
-            clearQueryCache(queryClient, ["enums", "metadatasettypes"]);
-            if (result.status !== 204) {
-                const redirectUrl = (await result.json())?.["redirect_uri"];
-                if (redirectUrl) {
-                    console.log(redirectUrl);
-                    window.location.replace(redirectUrl);
-                    return;
+        try {
+            const result = await apiFetch(`/api/logout`, {
+                method: "POST",
+                headers: { "Content-Type": "application/json" },
+                body: JSON.stringify(body),
+            });
+            if (result.ok) {
+                clearQueryCache(queryClient, ["enums", "metadatasettypes"]);
+                if (result.status !== 204) {
+                    const redirectUrl = (await result.json())?.["redirect_uri"];
+                    if (redirectUrl) {
+                        console.log(redirectUrl);
+                        window.location.replace(redirectUrl);
+                        return;
+                    }
                 }
+                setAuthenticated(false);
+            } else {
+                setNetworkError(true);
             }
-            setAuthenticated(false);
+        } catch (error) {
+            console.log("Network Error: ", error);
+            setNetworkError(true);
         }
     }
 
     // Check if already signed in
     // Since both apiInfo and a loggedin status are required to render main app, we'll query both in sequence to prevent unnecessary rerenders/reroutes
     useEffect(() => {
         (async () => {
-            const loginResult = await apiFetch(`/api/login`, { method: "POST" });
-            if (loginResult.ok) {
-                const loginInfo = await loginResult.json();
-                setCurrentUser(loginInfo);
-            }
-            setAuthenticated(loginResult.ok);
-
             let endpoints: LabSelection[] = [];
-            const availibleEndpoints = await fetch("/labs.json");
-            if (availibleEndpoints.ok) {
-                endpoints = await availibleEndpoints.json();
+            try {
+                const availibleEndpoints = await fetch("/labs.json");
+                if (availibleEndpoints.ok) {
+                    endpoints = await availibleEndpoints.json();
+                } else if (availibleEndpoints.status !== 404) {
```

Comment:
Can you correct the spelling typo here?

Reply:
This variable was spelled originally like this to not repeat the state `availableEndpoints`.